system/install-service.sh.in

```diff
@@ -0,0 +1,671 @@
+  [ -n "${NETDATA_INSTALLER_START_CMD}" ] && export NETDATA_INSTALLER_START_CMD
+}
+
+# =====================================================================
+# Help functions
+
+usage() {
+  cat << HEREDOC
+USAGE: install-service.sh [options]
+       where options include:
+
+  --source      Specify where to find the service files to install (default ${SVC_SOURCE}).
+  --type        Specify the type of service file to install. Specify a type of 'help' to get a list of valid types for your platform.
+  --cmds        Additionally print a list of commands for starting and stopping the agent with the detected service type.
+  --export-cmds Export the variables that would be printed by the --cmds option.
+  --cmds-only   Don't install, just handle the --cmds or --export-cmds option.
+  --enable      Explicitly enable the service on install (default is to enable if not already installed).
+  --disable     Explicitly disable the service on install.
+  --help        Print this help information.
+HEREDOC
+}
+
+help_types() {
+  cat << HEREDOC
+Valid service types for ${PLATFORM} are:
+$(valid_types)
+HEREDOC
+}
+
+# =====================================================================
+# systemd support functions
+
+issystemd() {
+  pids=''
+  p=''
+  myns=''
+  ns=''
+  systemctl=''
+
+  # if the directory /lib/systemd/system OR /usr/lib/systemd/system (SLES 12.x) does not exit, it is not systemd
+  if [ ! -d /lib/systemd/system ] && [ ! -d /usr/lib/systemd/system ]; then
+    return 1
+  fi
+
+  # if there is no systemctl command, it is not systemd
+  systemctl=$(command -v systemctl 2> /dev/null)
+  if [ -z "${systemctl}" ] || [ ! -x "${systemctl}" ]; then
+    return 1
+  fi
+
+  # if pid 1 is systemd, it is systemd
+  [ "$(basename "$(readlink /proc/1/exe)" 2> /dev/null)" = "systemd" ] && return 0
+
+  # if systemd is not running, it is not systemd
+  pids=$(safe_pidof systemd 2> /dev/null)
+  [ -z "${pids}" ] && return 1
+
+  # check if the running systemd processes are not in our namespace
+  myns="$(readlink /proc/self/ns/pid 2> /dev/null)"
+  for p in ${pids}; do
+    ns="$(readlink "/proc/${p}/ns/pid" 2> /dev/null)"
+
+    # if pid of systemd is in our namespace, it is systemd
+    [ -n "${myns}" ] && [ "${myns}" = "${ns}" ] && return 0
+  done
+
+  # else, it is not systemd
+  return 1
+}
+
+check_systemd() {
+  if [ -z "${IS_SYSTEMD}" ]; then
+    issystemd
+    IS_SYSTEMD="$?"
+  fi
+
+  return "${IS_SYSTEMD}"
+}
+
+get_systemd_service_dir() {
+  if [ -w "/lib/systemd/system" ]; then
+    echo "/lib/systemd/system"
+  elif [ -w "/usr/lib/systemd/system" ]; then
+    echo "/usr/lib/systemd/system"
+  elif [ -w "/etc/systemd/system" ]; then
+    echo "/etc/systemd/system"
+  else
+    error "Unable to detect systemd service directory."
+    exit 4
+  fi
+}
+
+install_systemd_service() {
+  SRCFILE="${SVC_SOURCE}/netdata.service"
+
+  if [ "$(systemctl version | head -n 1 | cut -f 2 -d ' ')" -le 235 ]; then
```

Comment:
This is failing, should it be `systemctl --version` ?

Reply:
Ah, yes, nice catch.